libmamba/src/core/subdirdata.cpp

```diff
@@ -695,4 +695,109 @@ namespace mamba
             fs::remove(m_solv_fn);
         }
     }
+
+    tl::expected<Jlap, std::runtime_error> Jlap::parse(const fs::u8path& path)
+    {
+        Jlap jlap_doc;
+        std::ifstream jlap_file = open_ifstream(path);
+        if (!jlap_file.is_open())
+        {
+            return tl::unexpected(std::runtime_error("No such file or directory"));
+        }
+
+        std::vector<std::string> lines;
+        std::string line;
+        while (std::getline(jlap_file, line))
```

Comment:
BTW we don't have already some utility function to do this? (read a file as lines into a vector)